### ISSUE TITLE
build rust package with non crates.io dependencies

### ISSUE THREAD

**Issue by flosse** [NixOS/nixpkgs#11237]
@madjar @cstrahan @wizeman @globin @Havvy @wkennington 

I'm trying to use `buildRustPackage` with non crates.io dependencies without success.
In my `Cargo.toml` there are dependencies that point to repositories, e.g.

```toml
[dependencies.rusted_cypher]
version = "0.7.1"
git = "git://github.com/livioribeiro/rusted-cypher.git"
branch = "feature/rustc-serialize"
features = ["rustc-serialize"]  
```
This leads to `fatal: Not a git repository (or any of the parent directories): .git` :(
How can I fix this problem?

**Comment by wizeman** [comment]
Ok, I've fixed this in 799f0f1f239b00177b72322e261ab350647dbd60 (pushed to master branch, should be available in the unstable channel in next update).

It was due to the slash (`/` character) in the branch name. It caused cargo to create the `.git` directory in the git checkout one level deeper in the filesystem tree.

**Comment by flosse** [comment]
> Ok, I've fixed this

great!

> should be available in the unstable channel

Is there a way to use it in the stable channel?
